src/core/qgssnappingconfig.h

```diff
@@ -52,14 +51,55 @@ class CORE_EXPORT QgsSnappingConfig
 
     /**
      * SnappingType defines on what object the snapping is performed
+     * \deprecated since QGIS 3.12 use SnappingTypeFlag instead.
      */
     enum SnappingType
     {
       Vertex = 1, //!< On vertices only
       VertexAndSegment = 2, //!< Both on vertices and segments
       Segment = 3, //!< On segments only
     };
-    Q_ENUM( SnappingType )
+    // TODO QGIS 4: remove
+    // this could not be tagged with Q_DECL_DEPRECATED due to Doxygen warning
+    // might be fixed in newer Doxygen (does not on 1.8.15, might be ok on 1.8.16)
+
+    /**
+     * SnappingTypeFlag defines on what object the snapping is performed
+     * \since QGIS 3.12
+     */
+    enum SnappingTypes
```

Comment:
you couldn't use the scope based enum?
This would have let you skip the `Flag` term to all enums.

Reply:
Oh sorry!
It worked so well that I forgot about your comment :smile: 
It's done.Thank you

Reply:
SIP is unhappy, since Vertex and Segment already exists:
```sip: core/auto_generated/qgssnappingconfig.sip:45: There is already an enum member in scope with the same Python name```

Reply:
And with a scoped enum, Travis is unhappy too since we use Qt 5.9. I think it's the bug described [here](https://bugreports.qt.io/browse/QTBUG-47652?jql=text%20~%20%22scoped%20enum%22

It's work on my machines (FreeBSD with Qt 5.13 and Arch/Manjaro Qt 5.14) and on the @SebastienPeillet (Ubuntu Qt 5.12)

Reply:
OK, sorry for the time lost.
It really look like that moving to scoped enums has to be postponed to QGIS 4.
It's a pity because the scoped enums are looking OK in the PyQGIS docs while the others are....verrrrrryyyy bad looking.

Reply:
OK. I reverted my two commits.
Since we use QT 5.9 we can't use scoped enum with flags.